### PR TITLE
Feature/387/change first step according to figma

### PR DESCRIPTION
**Pull request by Tolik170**
Changed first step according to figma, used API with countries and cities:

image
image image

### CODE REVIEW

File: src/services/location-service.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import axios from 'axios'
```
Comment (MickhaelSh) [inline src/services/location-service.js:1]: move this API call to backend part

File: src/containers/tutor-home-page/general-info/GeneralInfo.js [hunk 2]
```diff
@@ -1,71 +1,163 @@
...
+import img from '~/assets/img/tutor-home-page/become-tutor/general-info.svg'
+import { useStepContext } from '~/context/step-context'
+import { validations } from '~/containers/tutor-home-page/constants'
 import { styles } from '~/containers/tutor-home-page/general-info/general-info.styles'
 
-const GeneralInfo = ({ data, handleChange, handleBlur, errors, btnsBox, stepLabel }) => {
+const GeneralInfo = ({ btnsBox, stepLabel }) => {
   const { t } = useTranslation()
+  const { isDesktop, isMobile } = useBreakpoints()
+  const [counterColor, setCounterColor] = useState('#263238')
```
Comment (MickhaelSh) [inline src/containers/tutor-home-page/general-info/GeneralInfo.js:25]: [error, setError] =useState(false) and move color to palette

File: src/containers/tutor-home-page/constants.js [hunk 3]
```diff
@@ -3,18 +3,16 @@ import { nameField, textField } from '~/utils/validations/common'
 export const initialValues = {
   firstName: '',
   lastName: '',
-  country: '',
-  city: '',
+  country: null,
+  city: null,
   confirmAge: false,
```
Comment (MickhaelSh) [inline src/containers/tutor-home-page/constants.js:8]: remove confirmAge

File: src/containers/tutor-home-page/general-info/GeneralInfo.js (hunk 2 above)
Comment (Tolik170) [reply src/containers/tutor-home-page/general-info/GeneralInfo.js]: fixed

File: src/containers/tutor-home-page/constants.js (hunk 3 above)
Comment (Tolik170) [reply src/containers/tutor-home-page/constants.js]: fixed

File: src/containers/tutor-home-page/general-info/GeneralInfo.js [hunk 4]
```diff
@@ -1,71 +1,163 @@
...
+      })()
+    }
+
+    return () => {
+      setFetched(false)
+    }
+  }, [data.country, fetchCities])
+
+  useEffect(() => {
+    handleStepData(stepLabel, { ...data }, { ...errors })
```
[the remark this answers is not in the document]
Comment (Tolik170) [reply src/containers/tutor-home-page/general-info/GeneralInfo.js]: fixed

File: src/context/step-context.js [hunk 5]
```diff
@@ -1,10 +1,11 @@
 import { createContext, useCallback, useContext, useState } from 'react'
 import { stepLabels } from '~/containers/tutor-home-page/constants'
+import { initialValues } from '~/containers/tutor-home-page/constants'
 
 const StepContext = createContext()
 
 const StepProvider = ({ children }) => {
-  const [generalData, setGeneralData] = useState({})
+  const [generalData, setGeneralData] = useState({ data: { ...initialValues } })
```
[the remark this answers is not in the document]
Comment (Tolik170) [reply src/context/step-context.js]: fixed

File: src/components/app-auto-complete/AppAutoComplete.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+import Loader from '../loader/Loader'
+
+const defaultFilterOptions = (options, state) => {
+  const filterOptions = createFilterOptions()
+  return filterOptions(options, state)
+}
+
+const AppAutoComplete = ({
+  disableOption = false,
```
[the remark this answers is not in the document]
Comment (Tolik170) [reply src/components/app-auto-complete/AppAutoComplete.js]: fixed

File: src/services/location-service.js (hunk 1 above)
Comment (Tolik170) [reply src/services/location-service.js]: fixed

**Review by MickhaelSh — APPROVED** [review]

**Review by KhrystynaPavlikovska — APPROVED** [review]

**Review by Yurenko — APPROVED** [review]

**Review by maarekua — APPROVED** [review]

**Pull request merged**